language/src/modules/Language.rb

```diff
@@ -1437,12 +1405,70 @@ def readonly
       ProductFeatures.GetBooleanFeature("globals", "readonly_language")
     end
 
-    def locale_to_install
+    # Returns the locale to use, default or chosen
+    #
+    # Based on the readonly_language feature
+    #
+    # @return [String] default language when language set to read-only; chosen language otherwise
+    def locale
       language = readonly ? DEFAULT_FALLBACK_LANGUAGE : @language
 
       GetLocaleString(language)
     end
 
+    # Sets locale settings for given locale
+    #
+    # @param locale [String]
+    def prepare_locale_settings(locale)
+      @localed_conf ||= {}
+      @localed_conf["LANG"] = locale
+
+      fix_lc_messages(locale)
+
+      log.info("Locale: #{@localed_conf}")
+    end
+
+    # Fix the value of LC_MESSAGES for zh_HK locale
+    #
+    # It must be zh_TW
+    #
+    # @param locale [String]
+    def fix_lc_messages(locale)
+      @localed_conf ||= {}
+
+      if locale.start_with?("zh_HK")
+        @localed_conf["LC_MESSAGES"] = "zh_TW"
+      elsif @localed_conf["LC_MESSAGES"] == "zh_TW"
+        @localed_conf.delete("LC_MESSAGES")
+      end
+    end
+
+    # Returns the locale settings as args for localectl
+    #
+    # @return [String] a comma separated locale settings string; i.e, LANG=zh_HK,LC_MESAGES=zh_TW
+    def localectl_args
+      @localed_conf.map { |k, v| "#{k}=#{v}" }.join(",")
+    end
+
+    # Returns the command to make language settings persistent
+    #
+    # It is different depending on the stage, since for a not installed system it is needed to use
+    # the `systemd-firsrboot` tool, which does not work in a chroot
+    #
+    # @param locale [String]
+    #
+    # @return [String] the command to be executed
+    def save_command_for(locale)
+      if Stage.initial
+        # do use --root option, running in chroot does not work
+        "/usr/bin/systemd-firstboot --root #{Installation.destdir.shellescape} --locale #{locale.shellescape}"
```

Comment:
if you want to use it for cheetah, simply use array like `["/usr/bin/systemd-firstboot", "--root", Installation.destdir, "--locale",  locale]`

Reply:
Thank you Josef, I though that it will be the same, but your previous clarification allowed me to realize that it is not.